crates/matrix-sdk-base/src/store/integration_tests.rs

```diff
@@ -1,4 +1,823 @@
-//! Macro of integration tests for StateStore implementations.
+//! Trait and macro of integration tests for StateStore implementations.
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use async_trait::async_trait;
+use matrix_sdk_test::test_json;
+use ruma::{
+    api::client::media::get_content_thumbnail::v3::Method,
+    event_id,
+    events::{
+        presence::PresenceEvent,
+        receipt::{ReceiptThread, ReceiptType},
+        room::{
+            member::{
+                MembershipState, RoomMemberEventContent, StrippedRoomMemberEvent,
+                SyncRoomMemberEvent,
+            },
+            power_levels::RoomPowerLevelsEventContent,
+            topic::{OriginalRoomTopicEvent, RedactedRoomTopicEvent, RoomTopicEventContent},
+            MediaSource,
+        },
+        AnyEphemeralRoomEventContent, AnyGlobalAccountDataEvent, AnyRoomAccountDataEvent,
+        AnyStrippedStateEvent, AnySyncEphemeralRoomEvent, AnySyncStateEvent,
+        GlobalAccountDataEventType, RoomAccountDataEventType, StateEventType,
+    },
+    mxc_uri, room_id,
+    serde::Raw,
+    uint, user_id, EventId, OwnedEventId, RoomId, UserId,
+};
+use serde_json::{json, value::Value as JsonValue};
+
+use super::DynStateStore;
+use crate::{
+    deserialized_responses::MemberEvent,
+    media::{MediaFormat, MediaRequest, MediaThumbnailSize},
+    store::{Result, StateStoreExt},
+    RoomInfo, RoomType, StateChanges,
+};
+
+/// `StateStore` integration tests.
+///
+/// This trait is not meant to be used directly, but will be used with the [``]
+/// macro.
+#[async_trait]
```

Comment:
You need this ugly thing:

```suggestion
#[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
```